data-prepper-plugins/otel-trace-group-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OTelTraceGroupPrepper.java

```diff
@@ -88,6 +93,7 @@ public OTelTraceGroupPrepper(final PluginSetting pluginSetting) {
                 }
             } else {
                 recordsOut.add(record);
+                spanMissingTraceGroupCounter.increment();
                 final String spanId = (String) rawSpanMap.get(OTelTraceGroupPrepperConfig.SPAN_ID_FIELD);
                 LOG.info("Failed to find traceGroup for spanId: {} due to traceGroup missing for traceId: {}", spanId, traceId);
```

Comment:
Not part of this PR, however I just noticed we might want this to be a WARN instead of INFO.